src/main/java/com/cedarsoftware/util/io/factory/AbstractTemporalFactory.java

```diff
@@ -21,11 +21,11 @@ public T newInstance(Class<?> c, JsonObject<String, Object> job) {
         Object value = job.get("value");
 
         if (value instanceof String) {
-            return (T) fromString((String) value);
+            return fromString((String) value);
         }
 
         if (value instanceof Number) {
-            return (T) fromNumber((Number) value);
+            return fromNumber((Number) value);
```

Comment:
Can we call here to a method .getValue() on the jsonObject instance?  That method would get the "value" field and check it for all logical primitive types and convert it.  That way, any consumer of jsonObject instance that represents a value, can just call .getValue() and have the proper value returned (String, Date, Boolean, Character, Double, Float, Integer, Short, Long, BigDecimal, BigInteger, AtomicLong, AtomicInteger, AtomicBoolean)